homeassistant/components/intesishome/climate.py

```diff
@@ -135,29 +146,50 @@ def __init__(self, ih_device_id, ih_device, controller):
         self._min_temp = None
         self._target_temp = None
         self._outdoor_temp = None
+        self._hvac_mode = None
+        self._preset = None
+        self._preset_list = [PRESET_ECO, PRESET_COMFORT, PRESET_BOOST]
         self._run_hours = None
         self._rssi = None
-        self._swing = None
-        self._swing_list = None
+        self._swing_list = [SWING_OFF]
         self._vvane = None
         self._hvane = None
         self._power = False
         self._fan_speed = None
-        self._hvac_mode = None
-        self._fan_modes = controller.get_fan_speed_list(ih_device_id)
-        self._support = SUPPORT_TARGET_TEMPERATURE | SUPPORT_FAN_MODE
-        self._swing_list = [SWING_OFF]
+        self._support = 0
+        self._power_consumption_heat = None
+        self._power_consumption_cool = None
+
+        # Setpoint support
+        if controller.has_setpoint_control(ih_device_id):
+            self._support |= SUPPORT_TARGET_TEMPERATURE
 
-        if ih_device.get("config_vertical_vanes"):
+        # Setup swing list
+        if controller.has_vertical_swing(ih_device_id):
             self._swing_list.append(SWING_VERTICAL)
-        if ih_device.get("config_horizontal_vanes"):
+        if controller.has_horizontal_swing(ih_device_id):
             self._swing_list.append(SWING_HORIZONTAL)
-        if len(self._swing_list) == 3:
+        if SWING_HORIZONTAL in self._swing_list and SWING_VERTICAL in self._swing_list:
             self._swing_list.append(SWING_BOTH)
-            self._support |= SUPPORT_SWING_MODE
-        elif len(self._swing_list) == 2:
+        if len(self._swing_list) > 1:
             self._support |= SUPPORT_SWING_MODE
 
+        # Setup fan speeds
+        self._fan_modes = controller.get_fan_speed_list(ih_device_id)
+        if self._fan_modes:
+            self._support |= SUPPORT_FAN_MODE
+
+        # Preset support
+        if ih_device.get("climate_working_mode"):
+            self._support |= SUPPORT_PRESET_MODE
+
+        # Setup HVAC modes
+        self._hvac_mode_list = [HVAC_MODE_OFF]
```

Comment:
Because some UI will order by the order in this list, it is recommended you add `HVAC_MODE_OFF` as last option.